oci/oci_transport.go

```diff
@@ -47,12 +54,24 @@ func ParseReference(reference string) (types.ImageReference, error) {
 			return nil, fmt.Errorf("Invalid tag %s", tag)
 		}
 	}
-	return NewReference(dir, tag), nil
+	return NewReference(dir, tag)
 }
 
 // NewReference returns an OCI reference for a directory and a tag.
-func NewReference(dir, tag string) types.ImageReference {
-	return ociReference{dir: dir, tag: tag}
+//
+// We do not expose an API supplying the resolvedDir; we could, but recomputing it
+// is generally cheap enough that we prefer being confident about the properties of resolvedDir.
+func NewReference(dir, tag string) (types.ImageReference, error) {
+	resolved, err := private.ResolvePathToFullyExplicit(dir)
+	if err != nil {
+		return nil, err
+	}
+	// This is necessary to prevent directory paths returned by PolicyConfigurationNamespaces
+	// from being ambiguous with values of PolicyConfigurationIdentity.
+	if strings.Contains(resolved, ":") {
+		return nil, fmt.Errorf("Invalid OCI reference %s:%s: path %s contains a colon", dir, tag, resolved)
```

Comment:
I think this ok now. Eventually this OCI isn't mean for _directory_ like output like `dir` - that means `oci_dest.go` will work exactly as `docker_img_dest.go` works (remote services). I have a TODO to create a new OCITar destination probably which will adhere to this and not allow colon in path.